### PR TITLE
F4xx SPI Support. Was too easy. didn't get enough time to hate myself and think of a pun

### PR DESCRIPTION
**Pull request by tmb5932**
Support for STM32F446 SPI

### CODE REVIEW

**Review by ol2764RIT — APPROVED** [review]
good job travis, cooking up bangers left and right bb

File: include/EVT/io/pin.hpp [hunk 1]
```diff
@@ -128,6 +128,10 @@ enum class Pin {
 #endif
 
 #ifdef STM32F4xx
+    SPI_SCK = PA_9,// D8
```
Comment (mjmagee991) [inline include/EVT/io/pin.hpp:131]: Why did you label two of them with their D values but not the other two?

File: include/EVT/manager.hpp [hunk 2]
```diff
@@ -52,7 +53,7 @@
     //    #include <EVT/io/platform/f4xx/PWMf4xx.hpp>
```
Comment (mjmagee991) [inline include/EVT/manager.hpp:53]: Delete PWM and I2C from here because they've already been implemented

File: src/io/platform/f4xx/SPIf4xx.cpp [hunk 3]
```diff
@@ -0,0 +1,281 @@
...
+SPI::SPIStatus SPIf4xx::read(uint8_t* bytes, uint8_t length) {
+    HAL_StatusTypeDef halStatus = HAL_SPI_Receive(&halSPI, bytes, length, EVT_SPI_TIMEOUT);
+
+    return halToSPIStatus(halStatus);
+}
+
+/**
+ * Converts HAL Status into EVT Core's own SPIStatus enum
+ *
+ * @param halStatus
```
Comment (mjmagee991) [inline src/io/platform/f4xx/SPIf4xx.cpp:269]:
```suggestion
 * @param[in] halStatus HAL-defined status enum value
```

**Review by mjmagee991 — APPROVED** [review]
A couple details to clean up but looks pretty good

File: include/EVT/io/pin.hpp (hunk 1 above)
Comment (tmb5932) [reply include/EVT/io/pin.hpp:131]: well... i meant to talk to you about that but clearly forgot. The other 2 don't have a D label. Only SPI1 has the capabilities to be only on D's, but SPI1's necessary pins are being used by GPIO by default, so it isn't usable normally. I'll talk to you about it at EVT tonight.

**Review by aclowmclaughlin — CHANGES_REQUESTED** [review]
Looks mostly good.

File: include/EVT/io/pin.hpp (hunk 1 above)
Comment (tmb5932) [reply include/EVT/io/pin.hpp]: SPI1 is now the default SPI being used. Going to test it on thurs but everything should be in order.

**Review by aclowmclaughlin — APPROVED** [review]
Sure looks like code! (Good job)

**Pull request merged**